pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -449,7 +482,35 @@ func addPullSecret(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime) error
 	return svc.SetDesiredKubeObject(secret, comp.GetName()+"-pull-secret")
 }
 
-func addInitContainer(values map[string]any, version string) error {
+func addServiceAccount(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime, values map[string]any) error {
+
+	customImagePullSecret := map[string]any{}
+	if comp.Spec.Parameters.Service.CustomizationImage.ImagePullSecretRef.Name != "" {
+		err := copyCustomImagePullSecret(comp, svc)
+		if err != nil {
+			return err
+		}
+		customImagePullSecret = map[string]any{
+			"name": customImagePullsecretName,
+		}
+	}
+
+	pullSecrets := []map[string]any{
+		{
+			"name": pullsecretName,
+		},
+		customImagePullSecret,
```

Comment:
What happens if no pull secret is specified in the claim? Does it add an invalid secret and break everything?

Reply:
The field would look like this:
```
imagePullSecrets:
- name: pullsecret
- {}
```

It's not pretty. but it works

Reply:
I've refactored that part, to not generate an empty `{}` if a pullSecret is not specified.